platform/bitcoin/client.go

```diff
@@ -26,22 +27,46 @@ func InitClient(URL string) Client {
 	}
 }
 
-func (c *Client) GetTransactions(address string) (transfers TransactionsList, err error) {
+func (c *Client) GetTransactions(address string) (transactions TransactionsList, err error) {
 	path := fmt.Sprintf("address/%s", address)
-	err = c.Request.Get(&transfers, c.URL, path, url.Values{
+	err = c.Request.Get(&transactions, c.URL, path, url.Values{
 		"details":  {"txs"},
 		"pageSize": {strconv.FormatInt(blockatlas.TxPerPage*4, 10)},
 	})
-	return transfers, err
+	return transactions, err
 }
 
-func (c *Client) GetTransactionsByXpub(xpub string) (transfers TransactionsList, err error) {
+func (c *Client) GetTransactionsByXpub(xpub string) (transactions TransactionsList, err error) {
 	path := fmt.Sprintf("v2/xpub/%s", xpub)
 	args := url.Values{
 		"pageSize": {strconv.FormatInt(blockatlas.TxPerPage*4, 10)},
 		"details":  {"txs"},
 		"tokens":   {"derived"},
 	}
-	err = c.Request.Get(&transfers, c.URL, path, args)
-	return transfers, err
+	err = c.Request.Get(&transactions, c.URL, path, args)
+	return transactions, err
+}
+
+func (c *Client) GetTransactionsByBlock(number int64, page int64) (block Block, err error) {
+	path := fmt.Sprintf("v2/block/%s", strconv.FormatInt(number, 10))
+	args := url.Values{
+		"page": {strconv.FormatInt(page, 10)},
+	}
+	err = c.Request.Get(&block, c.URL, path, args)
+	return block, err
+}
+
+func (c *Client) GetTransactionsByBlockChan(number int64, page int64, out chan Block, wg *sync.WaitGroup) {
+	wg.Add(1)
+	defer wg.Done()
+	block, err := c.GetTransactionsByBlock(number, page)
+	if err != nil {
+		return
```

Comment:
We need handler this error. My suggestion is to create another channel to handle errors. Or we can put the result inside a struct with the `block` object and the error.